apps/examples/wifi_manager_sample/wm_test_dns.c

```diff
@@ -0,0 +1,184 @@
+/****************************************************************************
+ *
+ * Copyright 2021 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License\n");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <sys/types.h>
+#include <stdio.h>
+#include <errno.h>
+#include <netdb.h>
+#include <semaphore.h>
+#include <wifi_manager/wifi_manager.h>
+#include <stress_tool/st_perf.h>
+#include "wm_test.h"
+#include "wm_test_utils.h"
+#include "wm_test_network.h"
+#include "wm_test_log.h"
+
+//if semaphore operation failed then it'll try again 10ms later
+#define WT_SEM_TRY_WAIT_US 10000
+#define TAG "[WT]"
+
+#define WM_TEST_SIGNAL							\
+	do {										\
+		while (sem_post(&g_wm_sem) != 0) {		\
+			usleep(WT_SEM_TRY_WAIT_US);			\
+		}										\
+		WT_LOG(TAG, "send signal");			\
+	} while (0)
+
+#define WM_TEST_WAIT									\
+	do {												\
+		WT_LOG(TAG, "wait signal");					\
+		while (sem_wait(&g_wm_sem) != 0) {				\
+			if (errno == EINTR) {						\
+				continue;								\
+			}											\
+			WT_LOGE(TAG, "sem wait fail %d", errno);	\
+		}												\
+	} while (0)
+
+static char *WM_AP_SSID;
+static char *WM_AP_PASSWORD;
+static sem_t g_wm_sem;
+
+static void wm_sta_connected(wifi_manager_result_e);
+static void wm_sta_disconnected(wifi_manager_disconnect_e);
+
+static wifi_manager_cb_s g_wifi_callbacks = {
+	wm_sta_connected,
+	wm_sta_disconnected,
+	NULL,
+	NULL,
+	NULL,
+};
+
+static char *g_host_list[] = {
+	"www.google.com",
+	"www.facebook.com",
+	"www.naver.com",
+	"www.daum.net",
+};
+
+wifi_manager_result_e g_connect_result = WIFI_MANAGER_SUCCESS;
+
+static void wm_get_apinfo(wifi_manager_ap_config_s *apconfig)
+{
+	strncpy(apconfig->ssid, WM_AP_SSID, strlen(WM_AP_SSID) + 1);
+	apconfig->ssid_length = strlen(WM_AP_SSID);
+	apconfig->ap_auth_type = WIFI_MANAGER_AUTH_UNKNOWN;
+	strncpy(apconfig->passphrase, WM_AP_PASSWORD, strlen(WM_AP_PASSWORD) + 1);
+	apconfig->passphrase_length = strlen(WM_AP_PASSWORD);
+	apconfig->ap_crypto_type = WIFI_MANAGER_CRYPTO_UNKNOWN;
```

Comment:
doesn't it need auth info?? and can it connect to ap with unknown crypto?

Reply:
I changed it to run test after a board is connected to an AP.